j5basic/Formatters.py

```diff
@@ -33,9 +33,21 @@ def __str__(self):
 @python_2_unicode_compatible
 class StrFormattedMixIn(object):
     """Mixin for formatting with a Python % string."""
+    def __init__(self, *args, **kwargs):
+        self._re_entered = False
 
     def __str__(self):
-        return self.format_str % self
+        # For some reason the IntFormatter reenters this function, causing
```

Comment:
@dee42 Would you mind looking at this one? I am confused by it...

Reply:
I'm also pretty confused by it. I don't know if the recursion was caused by futurize (which is quite possible since I applied a weird `python2_unicode_compatible` decorator) or if it is normal for ("%02d" % IntFormatter(323) ) to call `str(IntFormater(323))`, causing recursion.

Reply:
Did you get to the bottom of this?

Reply:
I'm not sure if there is that much to this.

The `%` operator seems to occasionally call `__str__` on the object's we pass to it. Since we are passing `self` this can cause an infinite loop, so we make the function "re-entrant" be calling `super().__str__` if `__str__` tries to call itself at some point.

Reply:
I think we should just get rid of `StrFormattedMixIn` and it's subclasses and get rid of the uses of them in the framework code. The `int` and `float` formatters don't seem to be used for anything important there, and I'd rather not have something like this in place.

Reply:
I deleted the framework code that used the Int and Float formatters https://gitlab.sjsoft.com/j5-product/j5-framework/merge_requests/8102

Reply:
What's holding this up